### PR TITLE
Simplify `F.softmax` test

### PR DESCRIPTION
**Pull request by aksub99**
Simplify F.softmax test using testing.FunctionTestCase

### CODE REVIEW

**Comment by aksub99** [comment]
I have opened this PR because of issues I face while trying to commit the suggested fixes to PR #6242 .(Caused by problems in my fork). Sorry for the inconvenience caused.

**Comment by Crissman** [comment]
LGTM. Jenkins, test this please.
